app/src/main/java/com/graddex/graddex/PokemonRecyclerAdapter.kt

```diff
@@ -9,43 +10,56 @@ import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import com.graddex.graddex.models.PokemonResponse_Result
 
-class PokemonRecyclerAdapter
-    : ListAdapter<PokemonResponse_Result, PokemonViewHolder>(
-    object : DiffUtil.ItemCallback<PokemonResponse_Result>() {
-        override fun areItemsTheSame(
-            oldItem: PokemonResponse_Result,
-            newItem: PokemonResponse_Result
-        ): Boolean {
-            return oldItem === newItem
-        }
+class PokemonRecyclerAdapter(val listener: OnPokemonClicked)
```

Comment:
if we were to pass through a function here, it might look like:

```suggestion
class PokemonRecyclerAdapter(val listener: (String) -> Unit)
```